doc/CONTRIBUTING.md

```diff
@@ -1,118 +1,115 @@
+- For a **Major Feature**, first open an issue and outline your proposal sothat it can be discussed. This process allowss us to better coordinate our efforts, prevent duplication of work, ad help you to craft so that it is successflly accepted into the project.
+- **Small Features** can be crafter and directly [submitted as Pull Request](#submission-guidelines).
 
-Briefly, bootstrap the build system with
+## <a name="branch-name"></a> Branch name convention
 
+As a convention inside the repository, we use the following as the way to name git branches:
 ```
-$ ./autogen.sh
+feature/#issue-number
 ```
+It make it easier to know the issue number associated with the pull request.
 
-to create the build configuration script and then build the software with
+Also, this project's development workflow is based on the [GitFlow branching strategy](https://nvie.com/posts/a-successful-git-branching-model/) that is very well encouraged by the community.
 
-```
-$ ./configure 
-$ make
-```
-
-After the software is built, the binary may be found in the subdirectiry `src`.
+## <a name="submission-guidelines"></a>Submission Guidelines
 
-The file 
+### Submitting an Issue
+Before you submit an issue, please search the issue tracker. An issue for your problem might already exist and the discussion might inform you of workarounds readily available.
 
-```
-src/ttsnake
-```
+You can file new issues directly on the [Github repository issues page](https://github.com/courselab/snaskii21/issues/new).
 
-is actually a wrapper script wich invokes the real binary 
+### Submitting a Pull Request (PR)
 
-```
-src/ttsnake.bin
-```
-
-with appropirate arguments to read data files from the local subdirectory 'scenes'.
-
-In order to install the software,
-
-```
-$ make install
-```
-This command will copy the file `src/ttsnake.bin` to the system binary path, 
-usually `usr/bin` and invoke a install hook to rename it `ttsnake`. The
-installed binary will read data from the system data path, usually `usr/share`.
+Before submitting your Pull Request, consider the following steps:
 
-To install the software in a different location, use 
+1. Search on the [pull request tracker](https://github.com/courselab/snaskii21/pulls) for open or closes pull requests related to the same issue.
+2. Be sure that an issue describes the problem you are solving. Discussing ideas or implementations inside the issue page might help to ensure that the implementations has fewer flaws.
+3. Fork the [courselab/snaskii21](https://github.com/courselab/snaskii21) repository.
+4. Inside your forked repository, [create a new git branch](branch-name) to start working on your changes:
+    ```
+    git checkout -b feature/#<issue-number> develop
+    ```
+5. Create your patch.
+6. Commit your changes using a descriptive commit message that should follow our [commit message stardard](#commit-messages):
+    ```
+    git commit -am "<your-commit-message>"
```

Comment:
The best practice would be to actually use `git add` instead, since `-a` will just add all already tracked files and ignore any untracked ones.

Reply:
It was supposed to be just an example of commit, but I'll add it